src/python/WMCore/WMSpec/WMWorkloadTools.py

```diff
@@ -319,6 +319,12 @@ def setArgumentsWithDefault(arguments, argumentDefinition):
     for argument in argumentDefinition:
         if argument not in arguments and "default" in argumentDefinition[argument]:
             arguments[argument] = argumentDefinition[argument]["default"]
+            
+    # set the Campaign default value to the same as AcquisitionEra if Campaign is not specified
+    if ("AcquisitionEra" in arguments) and \
+       isinstance(arguments["AcquisitionEra"], basestring) and \
+        not arguments.get("Campaign", ""):
+        arguments["Campaign"] = arguments["AcquisitionEra"]
```

Comment:
This line looks ugly :) How about:
```
if not arguments.get("Campaign"):
    if "AcquisitionEra" in arguments and isinstance(arguments["AcquisitionEra"], basestring):
```
?

I guess we cannot directly test `isinstance(arguments["AcquisitionEra"], basestring)` because during assignment this arg may not be there, right?